state/indexer/block/kv/kv.go

```diff
@@ -601,9 +704,30 @@ func (idx *BlockerIndexer) indexEvents(batch dbm.Batch, events []abci.Event, hei
 				if err := batch.Set(key, heightBz); err != nil {
 					return err
 				}
+				keyArray = appendToKeyArray(keyArray, key)
 			}
 		}
 	}
+	err := batch.Set(getEventsForHeightKey(height), keyArray)
+	return err
+}
 
-	return nil
+func (idx *BlockerIndexer) deleteEvents(height int64) error {
+	eventsKey := getEventsForHeightKey(height)
+	keyArray, err := idx.store.Get(eventsKey)
```

Comment:
I just realized that for indexed data, in old versions of comet, this will not exist. Therefore you still need to have a loop through the events if the error is that they key is not found. 

So below
```
  if err == ErrKeyNotFound{
      deleteEventsSequentially(height)
   } else {
          return err 
    }

```

Reply:
Hmm... good point. I would just point out that it's `store.Get`, so if a key is not found, there is no error, but just the returned value is nil